collector/collector.go

```diff
@@ -42,6 +43,14 @@ var (
 	float64Mantissa uint64 = 9007199254740992
 	wrapCounters           = kingpin.Flag("snmp.wrap-large-counters", "Wrap 64-bit counters to avoid floating point rounding.").Default("true").Bool()
 	srcAddress             = kingpin.Flag("snmp.source-address", "Source address to send snmp from in the format 'address:port' to use when connecting targets. If the port parameter is empty or '0', as in '127.0.0.1:' or '[::1]:0', a source port number is automatically (random) chosen.").Default("").String()
+
+	SnmpDuration = promauto.NewHistogramVec(
```

Comment:
Minor nit, let's not make this a public variable.

```suggestion
	snmpDuration = promauto.NewHistogramVec(
```

Reply:
Oh, right. I prefer to not allow metrics to cross package boundaries.

How about we we have a function like `collector.InitModuleMetrics(modules []string)`?

Reply:
Like this?
```
diff --git a/collector/collector.go b/collector/collector.go
index 12770d1..3f0b33b 100644
--- a/collector/collector.go
+++ b/collector/collector.go
@@ -44,7 +44,7 @@ var (
        wrapCounters           = kingpin.Flag("snmp.wrap-large-counters", "Wrap 64-bit counters to avoid floating point rounding.").Default("true").Bool()
        srcAddress             = kingpin.Flag("snmp.source-address", "Source address to send snmp from in the format 'address:port' to use when connecting targets. If the port parameter is empty or '0', as in '127.0.0.1:' or '[::1]:0', a source port number is automatically (random) chosen.").Default("").String()

-       SnmpDuration = promauto.NewHistogramVec(
+       snmpDuration = promauto.NewHistogramVec(
                prometheus.HistogramOpts{
                        Name: "snmp_collection_duration_seconds",
                        Help: "Duration of collections by the SNMP exporter",
@@ -89,6 +89,10 @@ func listToOid(l []int) string {
        return strings.Join(result, ".")
 }

+func InitModuleMetrics(auth, module string) {
+       snmpDuration.WithLabelValues(auth, module)
+}
+
 type ScrapeResults struct {
        pdus    []gosnmp.SnmpPDU
        packets uint64
diff --git a/main.go b/main.go
index 3830afd..196f440 100644
--- a/main.go
+++ b/main.go
@@ -155,7 +155,7 @@ func (sc *SafeConfig) ReloadConfig(configFile string) (err error) {
        // Initialize metrics.
        for auth := range sc.C.Auths {
                for module := range sc.C.Modules {
-                       collector.SnmpDuration.WithLabelValues(auth, module)
+                       collector.InitModuleMetrics(auth, module)
                }
        }
        sc.Unlock()
```

Reply:
I was more thinking `collector.InitModuleMetrics(sc.C.Auths, sc.C.Modules)`.